src/borg/archiver.py

```diff
@@ -1108,6 +1108,7 @@ def _delete_repository(self, args, repository):
                 return self.exit_code
             repository.destroy()
             logger.info("Repository deleted.")
+            SecurityManager.destroy(repository)
```

Comment:
like this, the security dir is only killed when the repository is removed.

outdenting this one level would also kill the security dir when using `--cache-only` - do we want that?